manage.py

```diff
@@ -129,12 +129,57 @@ def set_dev_site_name() -> None:
                 name="pythondiscord.local:8000"
             )
 
+    @staticmethod
+    def run_metricity_init() -> None:
+        """
+        Initilise metricity relations and populate with some testing data.
+
+        This is done at run time since other projects, like Python bot,
+        rely on the site initilising it's own db, since they do not have
+        access to the init.sql file to mount a docker-compose volume.
+        """
+        import psycopg2
+        from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+        from urllib.parse import urlsplit
+        # The database URL has already been validated in `wait_for_postgres()`
+        db_url_parts = urlsplit(os.environ["DATABASE_URL"])
+        db_connection_kwargs = {
+            "host": db_url_parts.hostname,
+            "port": db_url_parts.port,
+            "user": db_url_parts.username,
+            "password": db_url_parts.password,
+        }
+        # Connect to pysite first to create metricity db
+        conn = psycopg2.connect(
+            database=db_url_parts.path[1:],
+            **db_connection_kwargs
+        )
+        conn.set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
+
+        with conn.cursor() as cursor:
+            cursor.execute("SELECT 1 FROM pg_catalog.pg_database WHERE datname = 'metricity'")
+            exists = cursor.fetchone()
+            if exists:
+                # Assume metricity is already populated if it exists
+                return
```

Comment:
Can something be [done in SQL](https://stackoverflow.com/questions/18389124/simulate-create-database-if-not-exists-for-postgresql) to avoid having to switch connections?

Reply:
I came across that myself too, and the latter part of that answer became quite complicated. While I don't mind using it, this seems much easier to understand what's going on.

Reply:
Hmm, I might spend a little more time on this, might have a working solution.

Reply:
This should do it, actually much nicer to read the code now, just the sql file is a little more complicated to follow, not much of an issue though.
https://github.com/python-discord/site/pull/583/commits/8f4271f8a8113d175cb8680735ec352305cd47a0